ql/src/java/org/apache/hadoop/hive/ql/exec/ColumnStatsUpdateTask.java

```diff
@@ -294,9 +298,22 @@ private ColumnStatisticsDesc getColumnStatsDesc(String dbName,
   }
 
   private int persistColumnStats(Hive db) throws HiveException, MetaException, IOException {
-    List<ColumnStatistics> colStats = new ArrayList<>();
-    colStats.add(constructColumnStatsFromInput());
-    SetPartitionsStatsRequest request = new SetPartitionsStatsRequest(colStats);
+    ColumnStatistics colStats = constructColumnStatsFromInput();
+    ColumnStatisticsDesc colStatsDesc = colStats.getStatsDesc();
+    // We do not support stats replication for a transactional table yet. If we are converting
+    // a non-transactional table to a transactional table during replication, we might get
+    // column statistics but we shouldn't update those.
+    if (AcidUtils.isTransactionalTable(getHive().getTable(colStatsDesc.getDbName(),
```

Comment:
Once we support ACID table stats replication this condition would go away. (work.getColStats() != null) is true only in case of replication. This function doesn't have access to replicationSpec, so checking for replicationScope is not possible. We could pass replicationSpec down to this function through the task<-work<-UpdateTableColumnStats message but all of that will be unnecessary after we support ACID table stats. So as a temporary measure I have used work.getColStats().

Reply:
sounds good.